app/controllers/application_controller.rb

```diff
@@ -1,4 +1,20 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   include SessionsHelper
+
+  def logged_in_user
+    return if logged_in?
+    store_location
+    flash[:danger] = t "login_first"
+    redirect_to login_url
+  end
+
+  def correct_user
+    @user = User.find_by id: params[:id]
+    redirect_to root_url if @user.nil? || !current_user?(@user)
```

Comment:
chia 2 method ra em, gồm 2 xử lý với 2 tin nhắn flash cho ng dùng chứ
- load_user: với `find_by` để tìm kiến tồn tại user, flash not found nếu ko tìm thấy (tức là user bị nil)
- Sau khi có user tồn tại, thì check correct_user nhé
nếu fail thì cũng phải có flash thông báo chứ ko phải redirect to root thôi đâu

Reply:
da